content/admin_views.py

```diff
@@ -40,3 +45,126 @@ def participant_mark_winner(request, participant_pk):
     participant.is_winner = not participant.is_winner
     participant.save()
     return JsonResponse({})
+
+
+#############
+# Reporting #
+#############
+
+
+def report_index_page(request):
+    return render(request, 'admin/reports/index.html')
+
+
+# Challenge reports
+def report_challenge_exports(request):
+    response = HttpResponse(content_type='text/csv; charset=utf-8')
+    response['Content-Disposition'] = 'attachment;filename=export-' + str(timezone.now()) + '.csv'
+
+    if request.method == 'POST':
+        if request.POST.get('action') == 'EXPORT-CHALLENGE-SUMMARY':
+            if request.POST['date_from'] is '' or request.POST['date_to'] is '':
+                date_from = date_to = None
+            else:
+                date_from = request.POST['date_from']
+                date_to = request.POST['date_to']
+            SummaryDataPerChallenge.export_csv(response, date_from=date_from, date_to=date_to)
+            return response
+        elif request.POST.get('action') == 'EXPORT-CHALLENGE-QUIZ-SUMMARY':
+            SummaryDataPerQuiz.export_csv(response)
+            return response
+        elif request.POST.get('action') == 'EXPORT-CHALLENGE-PICTURE':
+            challenge_name = request.POST['picture-challenge-name']
+            ChallengeExportPicture.export_csv(response, challenge_name=challenge_name)
+            return response
+        elif request.POST.get('action') == 'EXPORT-CHALLENGE-QUIZ':
+            challenge_name = request.POST['quiz-challenge-name']
+            ChallengeExportQuiz.export_csv(response, challenge_name=challenge_name)
+            return response
+        elif request.POST.get('action') == 'EXPORT-CHALLENGE-FREETEXT':
+            challenge_name = request.POST['freetext-challenge-name']
+            ChallengeExportFreetext.export_csv(response, challenge_name=challenge_name)
+            return response
+    elif request.method == 'GET':
+        context = {
+            'quiz_challenges': list(Challenge.objects.filter(type=Challenge.CTP_QUIZ)),
+            'picture_challenges': list(Challenge.objects.filter(type=Challenge.CTP_PICTURE)),
+            'freetext_challenges': list(Challenge.objects.filter(type=Challenge.CTP_FREEFORM))
+        }
+
+        return render(request, 'admin/reports/challenges.html', context=context)
+
+
+# Goal reports
+def report_goal_exports(request):
+    response = HttpResponse(content_type='text/csv; charset=utf-8')
```

Comment:
Move the creation of this response to inside the `POST` block as it's only relevant for that case.